### PR TITLE
The moduler tool

### PR DESCRIPTION
**Pull request by ArcturusZhang**
This is a new tool used for searching version.go files in the SDK for tags, and push them to github.
I wrote a conceptual pipeline using the moduler tool [here](https://github.com/ArcturusZhang/test-azure-pipeline/blob/master/azure-pipelines.yaml)

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] The PR targets the `latest` branch.
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Can you add some tests?

**Comment by ArcturusZhang** [comment]
Rebased this branch to `refactor-versioner` because the result of this tool based on the result of versioner tool.

**Comment by ArcturusZhang** [comment]
> Can you add some tests?

Tests added, please have a look

File: tools/moduler/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	vprintln("Pushing new tags")
+	for _, tag := range newTags {
+		if err := addNewTag(tag); err != nil {
+			return err
+		}
+	}
+	return executeCommand("git", "push", "origin", "--tags")
+}
+
+func generateProfiles(profilePath string) error {
```
Comment (jhendrixMSFT) [inline tools/moduler/cmd/root.go]: This should happen as part of the build pipeline.  Each tool should be responsible for one job, then they can be composed via scripts etc.  See also https://en.wikipedia.org/wiki/Unix_philosophy

Comment (ArcturusZhang) [reply tools/moduler/cmd/root.go]: Sure, I will move this back to after_script.

**Pull request closed without merging**